server/server_test.go

```diff
@@ -0,0 +1,96 @@
+package main
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/allegro/bigcache"
+)
+
+var (
+	testBaseString = "http://bigcache.org"
+)
+
+func testCacheSetup() {
+	cache, _ = bigcache.NewBigCache(bigcache.Config{
+		Shards:             1024,
+		LifeWindow:         10 * time.Minute,
+		MaxEntriesInWindow: 1000 * 10 * 60,
+		MaxEntrySize:       500,
+		Verbose:            true,
+		HardMaxCacheSize:   8192,
+		OnRemove:           nil,
+	})
+}
+
+func TestGetWithNoKey(t *testing.T) {
+	testCacheSetup()
+	req := httptest.NewRequest("GET", testBaseString+"/api/v1/cache/", nil)
+	rr := httptest.NewRecorder()
+
+	getCacheHandler(rr, req)
+	resp := rr.Result()
+
+	if resp.StatusCode != 400 {
+		t.Logf("want: 400; got: %d", resp.StatusCode)
+		t.Fail()
+	}
+}
+
+func TestGetWithMissingKey(t *testing.T) {
+	testCacheSetup()
+	req := httptest.NewRequest("GET", testBaseString+"/api/v1/cache/doesNotExist", nil)
+	rr := httptest.NewRecorder()
+
+	getCacheHandler(rr, req)
+	resp := rr.Result()
+
+	if resp.StatusCode != 404 {
+		t.Logf("want: 404; got: %d", resp.StatusCode)
+		t.Fail()
+	}
+}
+
+func TestGetKey(t *testing.T) {
+	testCacheSetup()
+	req := httptest.NewRequest("GET", testBaseString+"/api/v1/cache/getKey", nil)
+	rr := httptest.NewRecorder()
+
+	// set something.
+	cache.Set("getKey", []byte("123"))
+
+	getCacheHandler(rr, req)
+	resp := rr.Result()
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		t.Log(err)
+		t.FailNow()
+	}
+
+	if string(body) != "123" {
+		t.Logf("want: 123; got: %s.\n\tcan't get existing key getKey.", string(body))
+		t.Fail()
+	}
+}
+
+func TestPutKey(t *testing.T) {
```

Comment:
Missing case when there is no key in a request.

Reply:
Fixed.